### PR TITLE
shimv2: Add tracing to shimv2

### LINKED ISSUE #903: shimv2: Add tracing for Kata shimv2 binary
Add tracing for Kata shimv2, which does not have tracing yet. 

1.x issue: https://github.com/kata-containers/runtime/issues/1807

### PR DESCRIPTION
**Pull request by cmaf**
Add trace functions to shimv2. Tracing starts in New() which is called
twice, so the tracer is created twice. Stop tracing in Shutdown(),
Delete(), and StartShim() which is called after the initial New().

Load runtime config in New() since tracing cannot start without value
for Trace.

Fixes #903

Signed-off-by: Chelsea Mafrica <chelsea.e.mafrica@intel.com>

### CODE REVIEW

**Comment by cmaf** [comment]
backport PR: https://github.com/kata-containers/runtime/pull/2980

Testing ci, having problems running locally

File: src/runtime/containerd-shim-v2/service.go [hunk 1]
```diff
@@ -278,7 +302,22 @@ func getTopic(e interface{}) string {
...
+	span, ctx := opentracing.StartSpanFromContext(ctx, name)
+	span.SetTag("source", "runtime")
+
+	return span, ctx
+}
+
 func (s *service) Cleanup(ctx context.Context) (_ *taskAPI.DeleteResponse, err error) {
+	span, _ := trace(s.ctx, "Cleanup")
+	defer span.Finish()
+
```
Comment (liubin) [inline src/runtime/containerd-shim-v2/service.go:320]: Should call `defer katautils.StopTracing(s.ctx)` here? The cleanup will run shim and exit after this function is finished.

**Review by liubin — APPROVED** [review]

**Review by jodh-intel — APPROVED** [review]
Thanks @cmaf !

File: src/runtime/containerd-shim-v2/service.go (hunk 1 above)
Comment (cmaf) [reply src/runtime/containerd-shim-v2/service.go:320]: @liubin In this case, we miss `Shutdown()` if it is called after

**Review by mxpv — APPROVED** [review]

**Pull request merged**